services/core/FileWatchPublisher/filewatchpublisher/agent.py

```diff
@@ -0,0 +1,159 @@
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# The views and conclusions contained in the software and documentation
+# are those of the authors and should not be interpreted as representing
+# official policies, either expressed or implied, of the FreeBSD
+# Project.
+#
+# This material was prepared as an account of work sponsored by an
+# agency of the United States Government.  Neither the United States
+# Government nor the United States Department of Energy, nor Battelle,
+# nor any of their employees, nor any jurisdiction or organization that
+# has cooperated in the development of these materials, makes any
+# warranty, express or implied, or assumes any legal liability or
+# responsibility for the accuracy, completeness, or usefulness or any
+# information, apparatus, product, software, or process disclosed, or
+# represents that its use would not infringe privately owned rights.
+#
+# Reference herein to any specific commercial product, process, or
+# service by trade name, trademark, manufacturer, or otherwise does not
+# necessarily constitute or imply its endorsement, recommendation, or
+# favoring by the United States Government or any agency thereof, or
+# Battelle Memorial Institute. The views and opinions of authors
+# expressed herein do not necessarily state or reflect those of the
+# United States Government or any agency thereof.
+#
+# PACIFIC NORTHWEST NATIONAL LABORATORY
+# operated by BATTELLE for the UNITED STATES DEPARTMENT OF ENERGY
+# under Contract DE-AC05-76RL01830
+
+#}}}
+
+import logging
+import sys
+
+from datetime import datetime
+from volttron.platform.agent.utils import watch_file_with_fullpath
+from volttron.platform.vip.agent import Agent, RPC, Core
+from volttron.platform.agent import utils
+
+
+utils.setup_logging()
+_log = logging.getLogger(__name__)
+__version__ = '3.6'
+
+
+def file_watch_publisher(config_path, **kwargs):
+    """Load the FileWatchPublisher agent configuration and returns and instance
+    of the agent created using that configuration.
+
+    :param config_path: Path to a configuration file.
+
+    :type config_path: str
+    :returns: FileWatchPublisher agent instance
+    :rtype: FileWatchPublisher agent
+    """
+    config = utils.load_config(config_path)
+    return FileWatchPublisher(config, **kwargs)
+
+
+class FileWatchPublisher(Agent):
+    """Monitors files for changes and publishes added lines
+        on corresponding topics.
+
+    :param config: Configuration dict
+    :type config: dict
+
+    Example configuration:
+
+    .. code-block:: python
+
+        {
+	        "publish_file": [
+                {
+                    "file": "/var/log/syslog",
+                    "topic": "platforms/syslog",
+                },
```

Comment:
use platform/syslog as the the topic for these examples
I committed some of those above, but didn't in this one.